src/libstd/sys/unix/ext/thread.rs

```diff
@@ -0,0 +1,33 @@
+// Copyright 2015 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! Unix-specific extensions to primitives in the `std::process` module.
+
+use os::unix::raw::{pthread_t};
+use sys_common::{AsInner, IntoInner};
+use thread::{JoinHandle};
+
+#[unstable(feature = "thread_extensions", issue = "0")]
+pub type RawId = pthread_t;
+
+#[unstable(feature = "thread_extensions", issue = "0")]
+pub trait JoinHandleExt {
+    fn as_id(&self) -> &RawId;
```

Comment:
We may want to just call this `as_pthread_t` as it's going to be the same across all platforms.

Can you also add documentation for the trait and methods here as well?